src/lang-go.ts

```diff
@@ -727,63 +665,170 @@ export async function activateUsingWebSockets(ctx: sourcegraph.ExtensionContext)
+}
+
+const noopMaybeProviders = {
+    hover: () => Promise.resolve(undefined),
+    definition: () => Promise.resolve(undefined),
+    references: () => Promise.resolve(undefined),
+}
+
+/**
+ * Uses WebSockets to communicate with a language server.
+ */
+export async function initLSP(ctx: sourcegraph.ExtensionContext): Promise<MaybeProviders> {
+    const settings: BehaviorSubject<Settings> = new BehaviorSubject<Settings>({})
+    ctx.subscriptions.add(
+        sourcegraph.configuration.subscribe(() => {
+            settings.next(sourcegraph.configuration.get<Settings>().value)
+        })
+    )
+    const accessToken = await getOrTryToCreateAccessToken()
+    const langserverAddress = sourcegraph.configuration.get<Settings>().get('go.serverUrl')
+    if (!langserverAddress) {
+        return noopMaybeProviders
+    }
+
+    const unsubscribableSendRequest = mkSendRequest<any>(langserverAddress, accessToken)
+    const sendRequest = <T>(...args: Parameters<typeof unsubscribableSendRequest.sendRequest>): Promise<T> =>
+        unsubscribableSendRequest.sendRequest(...args)
+    ctx.subscriptions.add(unsubscribableSendRequest)
+
+    const hover = async (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) =>
+        convert.hover(
+            await sendRequest<LSP.Hover | null>({
+                rootURI: rootURIFromDoc(doc),
+                requestType: lsp.HoverRequest.type,
+                request: positionParams(doc, pos),
+                useCache: true,
+            })
+        )
+
+    const definition = async (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) =>
+        convert.xdefinition({
+            currentDocURI: doc.uri,
+            xdefinition: await sendRequest<lspext.Xdefinition[] | null>({
+                rootURI: rootURIFromDoc(doc),
+                requestType: new lsp.RequestType<any, any, any, void>('textDocument/xdefinition') as any,
+                request: positionParams(doc, pos),
+                useCache: true,
+            }),
+        })
+
+    const references = async (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) =>
+        convert.references({
+            currentDocURI: doc.uri,
+            references: await sendRequest<LSP.Location[]>({
+                rootURI: rootURIFromDoc(doc),
+                requestType: lsp.ReferencesRequest.type,
+                request: positionParams(doc, pos),
+                useCache: true,
+            }),
+        })
+
+    registerExternalReferences({ ctx, sendRequest, settings })
+    registerImplementations({ ctx, sendRequest })
+
+    return {
+        hover: wrapMaybe(hover),
+        definition: wrapMaybe(definition),
+        references: wrapMaybe(references),
+    }
+}
+
 function pathname(url: string): string {
     let pathname = url
     pathname = pathname.slice('git://'.length)
     pathname = pathname.slice(0, pathname.indexOf('?'))
     return pathname
 }
```

Comment:
The URL modules from different browsers sometimes behave differently. Renamed to `repoName`. Maybe this information could be passed around as JSON objects instead of as URLs.

Reply:
That is why we [polyfill `URL` in our code](https://sourcegraph.sgdev.org/sourcegraph/sourcegraph@08f418e90fb277fafeb331e717789e0aacbcb02d/-/blob/shared/src/polyfills.ts#L12-14), including the extension host (if this was a problem, we couldn't use `URL` anywhere, not just in sourcegraph-go). Did you observe the differences inside an extension?

Reply:
I think so, see https://github.com/sourcegraph/sourcegraph-go/blob/0cc85090acb4c3b07932ef5f8f4e35a3a9c761bd/src/lang-go.ts#L121

Reply:
That comment is 8 months old, which is likely before we added the global polyfill. In general, if a native API behaves inconsistently, the solution is to polyfill it and not work around it.